src/main/java/com/deepoove/swagger/diff/compare/PropertyDiff.java

```diff
@@ -12,75 +12,65 @@
+      Model rightModel = null == right ? null : newDefinitions.get(((RefProperty) right).getSimpleRef());
       String ref = leftModel != null
           ? ((RefProperty) left).getSimpleRef()
           : right != null
           ? ((RefProperty) right).getSimpleRef()
           : null;
-      ModelDiff diff = ModelDiff
-          .buildWithDefinition(oldDedinitions, newDedinitions)
-          .diff(leftModel, rightModel, ref);
-      increased.addAll(diff.getIncreased());
-      missing.addAll(diff.getMissing());
-      changed.addAll(diff.getChanged());
+      ModelDiffResult modelDiff = ModelDiff.build(oldDefinitions, newDefinitions).diff(leftModel, rightModel, ref);
+      diffResult.addIncreased(modelDiff.getIncreased());
+      diffResult.addMissing(modelDiff.getMissing());
+      diffResult.addChanged(modelDiff.getChanged());
+      diffResult.setHasOnlyCosmeticChanges(modelDiff.hasOnlyCosmeticChanges());
     } else if (left != null && right != null && !left.equals(right)) {
       ElProperty elProperty = new ElProperty();
       elProperty.setEl(String.format("%s -> %s", left.getType(), right.getType()));
       elProperty.setParentModelName("response");
       elProperty.setProperty(left);
       elProperty.setResponseTypeChanged(true);
-      changed.add(elProperty);
+      diffResult.addChanged(Collections.singleton(elProperty));
+      diffResult.setHasOnlyCosmeticChanges(hasOnlyCosmeticChanges(left, right));
     }
-    return this;
-  }
-
-  public List<ElProperty> getIncreased() {
-    return increased;
-  }
 
-  public void setIncreased(List<ElProperty> increased) {
-    this.increased = increased;
+    return diffResult;
   }
 
-  public List<ElProperty> getMissing() {
-    return missing;
+  public static boolean hasOnlyCosmeticChanges(Property left, Property right) {
+    return (areNotEqual(left.getDescription(), right.getDescription()) || areNotEqual(left.getExample(), right.getExample())) &&
+        areEqual(left.getAllowEmptyValue(), right.getAllowEmptyValue()) &&
+        areEqual(left.getAccess(), right.getAccess()) &&
+        areEqual(left.getTitle(), right.getTitle()) &&
+        areEqual(left.getReadOnly(), right.getReadOnly()) &&
+        areEqual(left.getName(), right.getName()) &&
+        areEqual(left.getType(), right.getType()) &&
+        areEqual(left.getFormat(), right.getFormat()) &&
+        areEqual(left.getVendorExtensions(), right.getVendorExtensions()) &&
+        areEqual(left.getPosition(), right.getPosition()) &&
+        left.getRequired() == right.getRequired();
   }
 
-  public void setMissing(List<ElProperty> missing) {
-    this.missing = missing;
+  private static boolean areNotEqual(Object left, Object right) {
+    return (left == null ^ right == null) || left != null && !left.equals(right);
   }
 
-  public List<ElProperty> getChanged() {
-    return changed;
+  private static boolean areEqual(Object left, Object right) {
+    return Objects.equals(left, right);
   }
```

Comment:
@zrrobbins @emm035 Is this easier to read? I could not only check for contract changes and compare with `isDiff()` because there is no `ChangedProperty` class that has an `isDiff()` method.

Reply:
I think just checking with `Objects.equals(l, r)` is fine. Why not just use `!Objects.equals(l, r)` for the inverse?

Reply:
I completely forgot to switch that out, don't even need a separate method.